tests/benchmark/ConvBench.cpp

```diff
@@ -0,0 +1,155 @@
+}
+
+/// Benchmark a convolution with specified parameters on square inputs.
+class ConvBench : public Benchmark {
+  /// Matrices
+  std::vector<float> outW;
+  std::vector<float> inW;
+  std::vector<float> filterW;
+  std::vector<float> biasW;
+
+  /// Dimensions
+  // [batch, x, y, channels]
+  size_t outWdims[4];
+  size_t inWdims[4];
+  // [outputChannels, x, y, inputChannels]
+  size_t filterWdims[4];
+
+  /// Parameters
+  size_t kernelSizes[2];
+  size_t strides[2];
+  size_t pads[2];
+  size_t group;
+  unsigned depthUnroll;
+
+
+public:
+  ConvBench(size_t inputBatch, size_t inputEdgeSize, size_t inputChannels, size_t filterMultiplier, 
+            size_t kernelSize, size_t stride, size_t pad, size_t group)
+      : kernelSizes{kernelSize, kernelSize}, strides{stride, stride}, 
+      pads{pad, pad}, group(group) {
+
+        inWdims[0] = inputBatch;  
+        inWdims[1] = inputEdgeSize;
+        inWdims[2] = inputEdgeSize;
+        inWdims[3] = inputChannels;
+
+        filterWdims[0] = filterMultiplier * group;
+        filterWdims[1] = kernelSize;
+        filterWdims[2] = kernelSize;
+        filterWdims[3] = inWdims[3] / group;
+
+        size_t outEdgeSize = ((inputEdgeSize + (2 * pad) - kernelSize) / stride) + 1;
+        outWdims[0] = inWdims[0];
+        outWdims[1] = outEdgeSize;
+        outWdims[2] = outEdgeSize;
+        outWdims[3] = filterWdims[0];
+
+        depthUnroll = (((outWdims[3] / group) % 8) == 0) ? 8 : 1; 
+      }
+
+  virtual void setup() override {
+    size_t outSize = mapMult(outWdims, 4);
+    size_t inSize = mapMult(inWdims, 4);
+    size_t filterSize = mapMult(filterWdims, 4);
+    size_t biasSize = filterWdims[0];
+
+    outW.resize(outSize);
+    inW.resize(inSize);
+    filterW.resize(filterSize);
+    biasW.resize(biasSize);
+
+    randomize(inSize, inW.data());
+    randomize(filterSize, filterW.data());
+    randomize(biasSize, biasW.data());
+  }
+
+  virtual void run() override {
+    // biasWDims isn't used in libjit_convolution_f, so we're passing NULL.
+    libjit_convolution_f(outW.data(), inW.data(), filterW.data(), biasW.data(), 
+                         outWdims, inWdims, filterWdims, NULL, 
+                         kernelSizes, strides, pads, group, depthUnroll);
+  }
+
+  virtual void teardown() override {}
+
+private:
+  size_t mapMult(size_t *vec, int size) {
+    size_t result = 1;
+    for (int i = 0; i < size; i++) {
+      result *= vec[i];
+    }
+    return result;
+  }
+
+  void randomize(size_t size, float *a) {
+    std::mt19937 gen;
+    std::uniform_real_distribution<> dis(-1.0, 1.0);
+    for (size_t i = 0; i < size; i++) {
+        a[i] = dis(gen);
+    }
+  }
+};
+
+int main() {
+  constexpr int reps = 10;
+  printf("inputBatch, inputEdgeSize, inputChannels, filterMultiplier, kernelSize, stride, pad, group, bestInSeconds\n");
+
+  for (size_t inputBatch : {1, 3}) {
+    for (size_t inputEdgeSize : {1, 224}) {
```

Comment:
7, 14, 28, and 56 are all interesting, especially with the higher channel numbers.  Can we get one of those in this loop?